### ISSUE TITLE
500 Error - POST /collectionSpecs

### ISSUE THREAD

**Issue by kixxauth** [oddnetworks/oddworks#173]
This error ends up being a 500 response when POSTing to `/collectionSpecs`

```
Error: spec.source String is required
at setItemSpec (/app/node_modules/@oddnetworks/oddworks/lib/services/catalog/commands/index.js:77:10)
```

https://github.com/oddnetworks/oddworks/blob/master/lib/services/catalog/commands/index.js#L76-L78

The desired behavior is to respond with a 422 from the controller before it makes it into the service level:

https://github.com/oddnetworks/oddworks/blob/master/lib/services/catalog/controllers/catalog-spec-list-controller.js#L51-L88

**Comment by kixxauth** [comment]
Additionally, when an unsupported source string is provided, a different 500 error is returned:

__POST body:__
```json

```
{
	"data": {
		"type": "collectionSpec",
		"attributes": {
			"channel": "nasa",
			"source": "foo"
		}
	} 
}
```

__respond body:__
```
{
  "errors": [
    {
      "status": "500",
      "title": "NotFoundError",
      "detail": "No handler for pattern role:provider,cmd:get,source:foo"
    }
  ]
}
```

This is expected, but the response code should be 422 and the error message should be more informative like "The provider source 'foo' is not supported".

**Comment by aljachimiak** [comment]
@kixxauth - please review.

I think that I introduced a regression and some testing messes in #200.

However, I believe the regressions have been addressed (#217) and the messes cleaned up (#218).

I think this issue can be closed now.

-Al

**Comment by kixxauth** [comment]
Addressing this in another PR 👍
